pwndbg/commands/context.py

```diff
@@ -377,21 +569,33 @@ def context_ghidra(target=sys.stdout, with_banner=True, width=None):
 
 
 @pwndbg.commands.ArgparsedCommand(parser, aliases=["ctx"], category=CommandCategory.CONTEXT)
-@pwndbg.commands.OnlyWhenRunning
 def context(subcontext=None) -> None:
     """
     Print out the current register, instruction, and stack context.
 
     Accepts subcommands 'reg', 'disasm', 'code', 'stack', 'backtrace', 'ghidra', 'args', 'threads', 'heap_tracker', 'expressions', and/or 'last_signal'.
     """
+    # Allow to view history after the program has exited
+    if not pwndbg.aglib.proc.alive and (context_history_size <= 0 or not context_history):
+        log.error("context: The program is not being run.")
+        return None
+
     if subcontext is None:
         subcontext = []
     args = subcontext
 
     if len(args) == 0:
         args = config_context_sections.split()
 
-    sections = [("legend", lambda *args, **kwargs: [M.legend()])] if args else []
+    def get_history_legend():
```

Comment:
Why a function at all?

Reply:
I mean, its called just once and in a condition; we can just do an if and assign proper value or not.

(note that we build the function in a function, which is done on each function call; meh)

Reply:
You're right, I was still in the mindset of capturing loop variables from the TUI window PR.